intercept.go

```diff
@@ -161,15 +162,16 @@ func InterceptMethod(intc InterceptorMethod, when When) {
 	})
 }
 
-func getInterceptors(when When, val reflect.Value) []*Interception {
-	result := []*Interception{}
+func getInterceptorEntries(when When, val reflect.Value) []*interceptionEntry {
+	result := []*interceptionEntry{}
```

Comment:
It's a small and simple optimization to reduce the calls to `findTarget()`, but is it possible to cache these entries on app start for further performance gains?